rpcs3/rpcs3qt/debugger_list.cpp

```diff
@@ -51,29 +51,38 @@ u32 debugger_list::GetCenteredAddress(u32 address) const
 	return address - ((m_item_count / 2) * 4);
 }
 
-void debugger_list::ShowAddress(u32 addr)
+void debugger_list::ShowAddress(u32 addr, bool force)
 {
 	auto IsBreakpoint = [this](u32 pc)
 	{
 		return m_breakpoint_handler->HasBreakpoint(pc);
 	};
 
-	if (xgui_settings->GetValue(gui::d_centerPC).toBool())
-	{
-		m_pc = GetCenteredAddress(addr);
-	}
-	else
+	const bool center_pc = xgui_settings->GetValue(gui::d_centerPC).toBool();
+
+	// How many spaces addr can move down without us needing to move the entire view
+	const u32 addr_margin = (m_item_count / (center_pc ? 2 : 1) - 4); // 4 is just a buffer of 4 spaces at the bottom
+
+	if (force || s32(addr) - m_pc > addr_margin * 4) // 4 is the number of bytes in each instruction
```

Comment:
there are 3 issues on this line:
- the cast to signed has no effect. (addr is casted to unsigned implicitly)
- c style casts are discouraged by our coding style, use static_cast.
- If you are going to use signed comparison atleast use s64 to ensure no overflow ever occurs.

Reply:
> the cast to signed has no effect. (addr is casted to unsigned implicitly)

you mean it's casted to _signed_ implicitly?
even then that's not true. both `addr` and `m_pc` are `u32` so it would do an unsigned subtraction which is of course not what i want (thought in practice it'd work because the result would be way bigger than `addr_margin * 4` would ever be)

> c style casts are discouraged by our coding style, use static_cast.
> If you are going to use signed comparison atleast use s64 to ensure no overflow ever occurs.

true, ok

Reply:
No, after its casted signed explicitly, its casted back to unsigned implicitly.
So this cast has no effect.
Im not sure what was the intention here so please explain.

Reply:
>No, after its casted to signed explicitly, its casted back to unsigned implicitly.
oh. because `addr_margin * 4` is unsigned?

the intention is to avoid overflow when ShowAddress gets called on an address that's above the one we're currently on (by ie. clicking on such a bp)

but as i said before, the overflow would cause the check to be true anyways since the number would be too big, so in practice it's fine even with no cast

so i'll just remove the cast i guess